lib/compat/wordpress-6.7/rest-api.php

```diff
@@ -48,8 +48,22 @@ function gutenberg_block_editor_preload_paths_6_7( $paths, $context ) {
 		if ( false !== $reusable_blocks_key ) {
 			unset( $paths[ $reusable_blocks_key ] );
 		}
-	}
 
+		/*
+		 * Removes the `context=edit` query parameter from the global styles preload path added in gutenberg_block_editor_preload_paths_6_6().
+		 * Uses array_filter because it may have been added again in WordPress Core 6.6 in src/wp-admin/edit-form-blocks.php
+		 * Reason: because GET requests to global styles items are accessible to roles with the `edit_posts` capability.
+		 */
+		$global_styles_edit_path = '/wp/v2/global-styles/' . WP_Theme_JSON_Resolver_Gutenberg::get_user_global_styles_post_id() . '?context=edit';
```

Comment:
Because read access permissions are looser, we don't need the `edit` response to be preloaded. The view/read is enough.

Reply:
Quickly hacking around to see the impact of these changes in relation to the [regression noted earlier](https://github.com/WordPress/gutenberg/pull/65071#pullrequestreview-2308168250), removing this change brings the flash of styling much closer to trunk for both admin and non-admin users. Simply including the view context path as well didn't appear to make much difference.

Reply:
I can reproduce. I'll remove this change 👍🏻 and test the preloading array in a follow up. Thanks for finding it.

Logically, all we need is the view preloaded, but I suspect there's something going on at a deeper level that's evading me.

Reply:
Yeah, I'm missing something here as well. Seems related to the fact we get either the edit or view context based on user caps. I was wondering if the preloaded paths needed the same conditional treatment.

That said, I'll need to re-test to see if the non-admin user faced the same issue before the latest commit.